server/src/main/java/org/elasticsearch/common/scheduler/SchedulerEngine.java

```diff
@@ -42,10 +42,16 @@ public class SchedulerEngine {
     public static class Job {
```

Comment:
Can `Job` and `Event` (and maybe some of the other things here) be proper `Record`s? If so, I'm indifferent to whether that's done on this PR or a subsequent PR, but I would like to see it done if it's possible to do.

Separately, I think `startTime` needs an `@Nullable` annotation and some commentary on **why** somebody should or should not set it. It worries me that there's this extra option with no manual.